### ISSUE TITLE
Wildcard '*' plugin circular loading

### ISSUE THREAD

**Issue by Luixo** [systemjs/systemjs#1137]
I wrote a simple plugin for systemjs that tries to fetch several urls based on recieved one in different folders specific to project and marked his with 'use for all' * wildcard:
```javascript
System.config({
	meta: {
		'*': {
			loader: 'wildcard-plugin.js'
		}
	}
});
```
It seems like on triggering first import, it hooks up the plugin (via metadata), then plugin tries to hook up himself and silently crash somewhere.
Unfortunately, i cannot import plugin before starting using it, so i cannot fix this painless way.
I have an option to load this plugin with other classifier, import something and than add wildcard classifier, but i think it's a hack.
My current solution is ugly too, i just check in meta extension module if i'm going to add module with loader to the loader itself (don't know how to reference code, so line 4605 in v0.19.23):
```javascript
if (!new RegExp(meta[module].loader + '$').test(name))
	extendMeta(load.metadata, meta[module], bestDepth != depth)
```
Is this behaviour intended and there is no beautiful way to add wildcard load plugin or it's just a bug?

**Comment by guybedford** [comment]
You would need to add the config - 

```javascript
System.config({
    meta: {
        '*': {
            loader: 'wildcard-plugin.js'
        },
        'wildcard-plugin.js': {
          loader: false
        }
    }
});
```

I don't think we want to do this automatically, as the configuration rules try to be consistent and understandable as much as possible with minimal "if" based branch. I've added https://github.com/systemjs/systemjs/commit/e2f76c2b10b966da91ddc427b4a9c7e8c2cc3d32 to at least catch this case though.

**Comment by Luixo** [comment]
Thanks for current version solution!

**Comment by guybedford** [comment]
Released in 0.19.24.
